src/util/terminology.cpp

```diff
@@ -0,0 +1,246 @@
+                                                   Poco::Net::Session::Ptr &session_ptr,
+                                                   const std::string &name,
+                                                   const std::string &terminology,
+                                                   const std::string &key,
+                                                   const std::string &match_type,
+                                                   const bool first_hit,
+                                                   const HandleNotResolvable &on_not_resolvable)
+{
+    Poco::URI uri("https://terminologies.gfbio.org/api/terminologies/search");
+    uri.addQueryParameter("query", name);
+    uri.addQueryParameter("terminologies", terminology);
+    uri.addQueryParameter("match_type", match_type);
+    if(first_hit)
+        uri.addQueryParameter("first_hit", "true");
+
+    Poco::Net::HTTPSClientSession session(uri.getHost(), uri.getPort(), context, session_ptr);
+    Poco::Net::HTTPRequest request(Poco::Net::HTTPRequest::HTTP_GET, uri.getPathAndQuery(), Poco::Net::HTTPRequest::HTTP_1_1);
+    Poco::Net::HTTPResponse response;
+
+    session.sendRequest(request);
+    std::istream& respStream = session.receiveResponse(response);
+
+    //read response stream into json object
+    Json::Value response_json;
+    if (response.getStatus() == Poco::Net::HTTPResponse::HTTP_OK)
+    {
+        respStream >> response_json;
+    } else {
+        response_json = Json::Value::null;
+    }
+
+    //retrieve wanted element from result json, if not valid result return not_resolved.
+    std::string not_resolved = (on_not_resolvable == EMPTY) ? "" : name;
+    if (response_json.isNull())
+    {
+        return std::make_pair(name, not_resolved);
+    } else
+    {
+        Json::Value results = response_json["results"];
+
+        Json::Value val = results[0].get(key, not_resolved);
+        if(val.isArray())
+            return std::make_pair(name, val[0].asString());
+        else
+            return std::make_pair(name, val.asString());
+    }
+}
+
+/**
+ * Same method as resolveSingleNameInternal, but also return the session_ptr to the sslSession of the created HttpsClientSession.
+ * @return a pair, first element is a string_pair as in resolveSingleNameInternal, second element is the Session::Ptr.
+ */
+Terminology::string_session_ptr_pair
+Terminology::resolveSingleNameSetSessionPtr(Poco::Net::Context::Ptr &context,
+                                            const std::string &name,
+                                            const std::string &terminology,
+                                            const std::string &key,
+                                            const std::string &match_type,
+                                            const bool first_hit,
+                                            const HandleNotResolvable on_not_resolvable) {
+
+    Poco::URI uri("https://terminologies.gfbio.org/api/terminologies/search");
```

Comment:
load url from config to be able to replace e.g. with dev version of terminology server